snuba/query/processors/logical/optional_attribute_aggregation.py

```diff
@@ -37,66 +37,69 @@ def __init__(
         self._curried_aggregation_names = curried_aggregation_names
 
     def process_query(self, query: Query, query_settings: QuerySettings) -> None:
-        def is_attribute_expression(exp_parameters: tuple[Expression, ...]) -> bool:
-            if len(exp_parameters) != 1:
-                return False
-            func_param = exp_parameters[0]
-            if not isinstance(func_param, SubscriptableReference):
-                return False
-            if func_param.column.column_name in self._attribute_column_names:
-                return True
-            return False
+        def find_subscriptable_reference(exp: Expression) -> SubscriptableReference:
```

Comment:
why do you need to do this? transform expressions will already iterate through every expression in the query

Reply:
this is to handle cases like `sum(multiply(multiply(...)))`